mock/mock.go

```diff
@@ -203,6 +203,14 @@ type Mock struct {
 	mutex sync.Mutex
 }
 
+// String provides a %v format string for Mock.
+// Note: this is used implicitly by Arguments.Diff if a Mock is passed.
+// It exists because go's default %v formatting traverses the struct
+// without acquiring the mutex, which is detected by go test -race.
+func (m *Mock) String() string {
```

Comment:
It is very strange to implement the stringer method through a pointer and not through a value. most often this method is implemented by value.

Reply:
@xorcare the Mock holds a mutex, so we can't pass by value or the race detector will detect the copy